sdk-tools/src/sign.rs

```diff
@@ -0,0 +1,272 @@
+            .arg("-config")
+            .arg(config_path)
+            .arg("-out")
+            .arg(output_datfile);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+
+    /// Combine an unsigned enclave and signature into the output enclave, after
+    /// checking the signature.
+    pub fn catsig(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        public_key_pem: &Path,
+        gendata_output: &Path,
+        signature: &Path,
+        output_enclave: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("catsig")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-key")
+            .arg(public_key_pem)
+            .arg("-unsigned")
+            .arg(gendata_output)
+            .arg("-sig")
+            .arg(signature)
+            .arg("-out")
+            .arg(output_enclave);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        cmd
+    }
+
+    /// Examine a signed enclave file and dump the data
+    pub fn dump(
+        &mut self,
+        signed_enclave: &Path,
+        css_file_path: &Path,
+        dump_file_path: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("dump")
+            .arg("-enclave")
+            .arg(signed_enclave)
+            .arg("-dumpfile")
+            .arg(dump_file_path)
+            .arg("-cssfile")
+            .arg(css_file_path);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+}
+
+/// Construct a new SgxSign utility around the given executable path
+impl From<PathBuf> for SgxSign {
+    fn from(sgx_sign_path: PathBuf) -> Self {
+        Self {
+            sgx_sign_path,
+            ignore_relocation_error: false,
+            ignore_init_section_error: false,
+            resign: false,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_generate_sign_command() {
+        let cmd = format!(
+            "{:?}",
+            SgxSign::default().sign(
+                Path::new("unsigned_enclave"),
+                Path::new("config_path"),
+                Path::new("private_key"),
+                Path::new("output_enclave"),
+            )
+        );
```

Comment:
:thought_balloon: This is probably sufficient for this logic, but I would suggest in the future having:
- Two happy path tests with different values, though unlikely it helps ensure someone didn't do something like hardcode "private_key" as the name of the key file
-  One test per each builder method, `allow_relocations()` to ensure they are properly added  to the command line
- One test with all of the builder methods just to show they play nicely together. This one requires pragmatism with the current 3 methods it's ok to do all, but if there were 5 or more it quickly becomes noisy to run them all in one test, but multiple should still be run.